packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,569 @@
+
+### Polling Response
+
+Each response back to FCL must be "wrapped" in a Polling Response. Each Polling Response can have it's status as `"APPROVED"`, `"DECLINED"`, or `"PENDING"`.
+
+It is entirely acceptible for your service to immediately return an `"APPROVED"` Polling Reponse, skipping a `"PENDING"` state.
+
+`"DECLINED"` Polling Responses must include a human readable reason for why it was declined.
+
+```javascript
+// APPROVED
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "APPROVED",
+  data: ___, // what the service needs to send to FCL
+}
+
+// Declined
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "DECLINED",
+  reason: "Declined by user."
+}
+
+// Pending - Simple
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // will be included in the requests body
+    params: {}, // will be included in the requests url
+  }
+}
+
+// Pending - First Time with Local
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // included in body of request
+    params: {}, // included as query params on endpoint
+  },
+  local: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "view/frame",
+    endpoint: "https://____", // the iframe that will be rendered,
+    data: {}, // sent to frame when ready
+    params: {}, // included as query params on endpoint
+  }
+}
+```
+
+#### data and params
+
+`data` and `params` are information that the wallet can provide in the service config that FCL will pass back to the service.
+- `params` will be added onto the `endpoint` as query params.
+- `data` will be included in the body of the `HTTP/POST` request or in the `FCL:VIEW:READY:RESPONSE` for a `IFRAME/RPC`, `POP/RPC` or `TAB/RPC`.
+
+# Authentication Service
+
+In the following examples, we'll walk you through the process of building an authentication service.
+
+In FCL, wallets are configured by passing in a wallet provider's authentication URL as the `discovery.wallet` config variable.
+
+As someone who is making an FCL compatible wallet, you will need to make and expose a webpage or API hosted at an authentication endpoint that FCL will use.
+
+```javascript
+// IN APPLICATION
+// configuring fcl to point at a wallet looks like this
+import {config} from "@onflow/fcl"
+
+config({
+  "discovery.wallet": "your-url-that-fcl-will-use-for-authentication",
+  "discovery.wallet.method": "IFRAME/RPC" // Available methods are "IFRAME/RPC", "POP/RPC", "TAB/RPC" or "HTTP/POST"
```

Comment:
Maybe mention as a comment that `discovery.wallet.method` is optional and defaults to `IFRAME/RPC` I know you specify it later, but people are mostly likely just going to be looking at this code snippet :P